src/plugins/core/sheet.ts

```diff
@@ -233,6 +256,39 @@ export class SheetPlugin extends CorePlugin<SheetState> implements SheetState {
     }
   }
 
+  /**
+   * Checks if all non-frozen header indices are present in the provided elements of selected rows/columns.
+   * This validation ensures that all rows or columns cannot be deleted or hidden when frozen panes exist.
+   */
+  isAllHeaderIndices(sheetId: string, dimension: string, elements: HeaderIndex[]): boolean {
+    const paneDivisions = this.getters.getPaneDivisions(sheetId);
+    const startIndex = dimension === "ROW" ? paneDivisions.ySplit : paneDivisions.xSplit;
+    const endIndex =
+      dimension === "ROW"
+        ? this.getters.getNumberRows(sheetId) - 1
+        : this.getters.getNumberCols(sheetId) - 1;
+
+    const hiddenColsRowsGroups: Set<HeaderIndex> = new Set(
+      dimension === "ROW"
+        ? this.getters.getHiddenRowsGroups(sheetId).flat()
```

Comment:
Just scrolled through the pr really fast, and this won't work because, as I noted in the pad, the headers can be hidden by UI features (data filters). I'm not 100% sur on the best way to proceed, but this definitely shouldn'tbe handled in core (or maybe as a temporary stopgag measure).

Reply:
Hi, I'm aware of this issue. Currently this code fixes the issue with delete and hide of rows/cols in normal cases. I'm trying to figure out a way to fix data filters, but it seems rather tricky.

Second point you mentioned that I should not include it in core, well freeze is originally controlled in sheet.ts, so it made more sense to modify the allowDispatch there. I'll try to look for more alternatives